### PR TITLE
Social Menus & Icons: Add Discord Support

### LINKED ISSUE #11093: Social Menu & Social Media Icons: Add Discord support
The only one I'm missing!

### PR DESCRIPTION
**Pull request by Aurorum**
Resolves #11093 

I'm not quite sure what the deal is with social icons. In testing these changes, this seems to work for social menus, but my understanding is that Automattic/jetpack#37549 would need to be addressed before it's possible to add a Discord social icon. I also don't really understand the CSS bit of this file, it doesn't seem necessary to add? Is that just for social icons too? Please correct me if that isn't the case! 

#### Changes proposed in this Pull Request:

Add Discord support to the social menu. `discord.com` is another URL, but the main use of Discord on a social menu is to provide an invite to a server, and that just redirects. The main two URLs for that are `discord.gg` and `discordapp.com`

screenshot_20190203-093216

#### Testing instructions:

Go to the Customizer with a theme that supports social menus and add a social menu using one of the two Discord links above, then verify the icon displays as expected. 

fgsddfgssfgdgfsd

#### Proposed changelog entry for your changes:

Social Menus: Add Discord support

### CODE REVIEW

File: modules/theme-tools/social-menu/icon-functions.php [hunk 1]
```diff
@@ -130,6 +130,8 @@ function jetpack_social_menu_social_links_icons() {
 			'behance.net'     => 'behance',
 			'codepen.io'      => 'codepen',
 			'deviantart.com'  => 'deviantart',
+			'discord.gg'      => 'discord',
+			'discordapp.com'  => 'discord',
```
Comment (thomasguillot) [inline modules/theme-tools/social-menu/icon-functions.php]: This looks great and I will be to approve it but would you mind adding Discord to `modules/widgets/social-icons.php` so the Social Icons Widget can also benefit from this addition.

**Review by thomasguillot — CHANGES_REQUESTED** [review]

File: modules/theme-tools/social-menu/icon-functions.php (hunk 1 above)
Comment (torres126) [reply modules/theme-tools/social-menu/icon-functions.php]: Thanks @thomasguillot, I wasn't sure if that could only be done once Discord had been added to the social icons repo, but it doesn't seem that's the case. I've added it now. :)

gsfsgffgsdgsfd

**Review by thomasguillot — DISMISSED** [review]
LGTM 👍

**Comment by torres126** [comment]
@mdawaffe The original SVG came from Discord's website but the sizing and viewbox seemed to cause severe issues, so I resized it and ran it through svgomg. Somewhere, that in turn seems for it to have become a few pixels out of centre. :(

Does this look better now? :) 

gsfdfsgdfgsdfsdg

In context:

fgsddfgssfgdgfsd

**Comment by mdawaffe** [comment]
I feel like the box part of the logo should be horizontally and vertically centered.

> The original SVG came from Discord's website

Is it licensed for use elsewhere?

**Comment by torres126** [comment]
I'll take a look at the vertical centering tomorrow! As for the license, the details are here: https://discordapp.com/branding

Looks like the folks over at Discord need to be contacted though. 

> Discord is a playful and fun brand that doesn't take itself too seriously. That said, we're keen on keeping things quality so if you're gonna use our art for something please keep it tasteful and send it our way for approval. 

Would you like me to pop them an email or is it something which I can't do?

**Comment by mdawaffe** [comment]
OK - their branding guidelines about vertical placement seems pretty clear - thanks for looking. Looks like the vertical spacing in this PR is fine.

> Would you like me to pop them an email or is it something which I can't do?

Thanks for the offer, but that's something we'll need to do.

**Comment by jeherve** [comment]
@thomasguillot Is it ok for us to add SVGs, but not Genericons? If a theme still uses the Genericon fallback, this won't work for them.

**Comment by thomasguillot** [comment]
> If a theme still uses the Genericon fallback, this won't work for them.

@jeherve correct. But that's already the case, the SVG supports more social networks than Genericon

**Review by jeherve — APPROVED** [review]
Alright! This should be good to merge then. All what's left is contacting Discord.

**Comment by mdawaffe** [comment]
Started the contact Discord process at p7xJch-10S-p2.

**Comment by mdawaffe** [comment]
Completed the contact Discord process at p9dueE-wU-p2.

**Comment by mdawaffe** [comment]
Deployed to WP.com finally (oops!) via D25673-code.

**Pull request merged**